### ISSUE TITLE
HashSet has no from_list constructor

### ISSUE THREAD

**Issue by ijt** [elixir-lang/elixir#4272]
It's surprising that the HashSet module has to_list but not from_list:

    iex(1)> HashSet.to_list(Enum.into([1], HashSet.new))   
    [1]
    iex(2)> HashSet.from_list [1]                                                
    ** (UndefinedFunctionError) undefined function HashSet.from_list/1
        (elixir) HashSet.from_list([1])
    iex(5)> HashSet.new [1]                              
    ** (UndefinedFunctionError) undefined function HashSet.new/1
        (elixir) HashSet.new([1])

The examples I've seen show Enum.into([1], HashSet.new), but it seems a bit wordy for such a simple thing.

**Comment by michalmuskala** [comment]
As far as I know `Enum.into/2` is the preferred way of constructing `HashSet`. Given it's deprecated as of Elixir 1.2 I doubt it there will be any new additions.

The newer `MapSet` features `MapSet.new/1` and `MapSet.new/2` that create the set from any enumerable.
